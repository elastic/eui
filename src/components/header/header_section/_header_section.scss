.euiHeaderSection {
  display: flex;
  flex-grow: 0;
  flex-shrink: 0;
}

.euiHeaderSection--grow,
.euiHeaderSection--left {
  flex-grow: 1;
}

.euiHeaderSection--dontGrow {
  flex-grow: 0;
}
