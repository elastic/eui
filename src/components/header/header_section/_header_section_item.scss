// Header items are the small icon links that pop up menus.

.euiHeaderSectionItem {
  position: relative;
  display: flex;
  align-items: center;

  &:after {
    position: absolute;
    content: '';
    top: $euiSize;
    bottom: 0;
    background: $euiBorderColor;
    left: 0;
  }
}

.euiHeaderSectionItem__button {
  height: $euiHeaderChildSize;
  min-width: $euiHeaderChildSize;
  text-align: center;
  font-size: 0; // aligns icons better vertically

  &:hover {
    background: $euiColorLightestShade;
  }

  &:focus {
    background: $euiFocusBackgroundColor;
  }
}

.euiHeaderSectionItem--borderLeft {
  &:after {
    left: 0;
    width: 1px;
  }
}

.euiHeaderSectionItem--borderRight {
  &:after {
    width: 1px;
    left: auto;
    right: 0;
  }
}

// SET FOR DEPRECATION: 2/21/20
// The `euiHeaderNotification` class was needed to be manually applied
// Now notifications can be automatically added to the buttons via props
.euiHeaderNotification,
.euiHeaderSectionItemButton__notification {
  position: absolute;
  top: 9%;
  right: 9%;
  box-shadow: 0 0 0 1px $euiHeaderBackgroundColor;
}

@include euiBreakpoint('xs') {
  .euiHeaderSectionItem,
  .euiHeaderSectionItem__button {
    min-width: $euiHeaderChildSize * .75;
  }

  .euiHeaderSectionItem--borderLeft,
  .euiHeaderSectionItem--borderRight {
    &:after {
      display: none;
    }
  }

  // On small screens just show a small dot indicating there are notifications
  .euiHeaderNotification,
  .euiHeaderSectionItemButton__notification {
    @include size($euiSizeS);
    top: 20%;
    min-width: 0;
    border-radius: $euiSizeS;
    color: $euiColorAccent;
    overflow: hidden;
  }
}
