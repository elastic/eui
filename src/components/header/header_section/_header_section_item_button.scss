.euiHeaderSectionItemButton {
  position: relative; // For positioning the notification
  height: $euiHeaderChildSize;
  min-width: $euiHeaderChildSize;
  text-align: center;
  font-size: 0; // aligns icons better vertically
}

.euiHeaderSectionItemButton__notification {
  position: absolute;

  &--dot {
    top: 0;
    right: 0;
    stroke: $euiHeaderBackgroundColor;
  }

  &--badge {
    top: 9%;
    right: 9%;
    box-shadow: 0 0 0 1px $euiHeaderBackgroundColor;
  }
}


.euiHeaderSectionItemButton__content {
  // This element is a span and we're changing the display because inline elements canâ€™t take a transform
  display: inline-block;
}

@include euiBreakpoint('xs') {
  .euiHeaderSectionItemButton {
    min-width: $euiHeaderChildSize * .75;
  }

  // Using specificity to override the default icon size
  .euiHeaderSectionItemButton__notification.euiHeaderSectionItemButton__notification--dot {
    @include size($euiSize);
    top: 9%;
  }
}
