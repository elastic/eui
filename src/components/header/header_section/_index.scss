@import 'header_section';
@import 'header_section_item';
@import 'header_section_item_button';