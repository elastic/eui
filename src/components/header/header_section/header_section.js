import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const sideToClassNameMap = {
  left: 'euiHeaderSection--left',
  right: 'euiHeaderSection--right',
};

const SIDES = Object.keys(sideToClassNameMap);

export const EuiHeaderSection = ({ side, children, className, ...rest }) => {
  const classes = classNames('euiHeaderSection', sideToClassNameMap[side], className);

  return (
    <div
      className={classes}
      {...rest}
    >
      {children}
    </div>
  );
};

EuiHeaderSection.propTypes = {
  side: PropTypes.oneOf(SIDES),
};

EuiHeaderSection.defaultProps = {
  side: 'left',
};
