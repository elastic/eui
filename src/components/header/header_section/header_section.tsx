/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { FunctionComponent, HTMLAttributes } from 'react';
import classNames from 'classnames';

import { CommonProps } from '../../common';

type HeaderSectionSide = 'left' | 'right';

const sideToClassNameMap: { [side in HeaderSectionSide]: string } = {
  left: 'euiHeaderSection--left',
  right: 'euiHeaderSection--right',
};

export type EuiHeaderSectionProps = CommonProps &
  HTMLAttributes<HTMLDivElement> & {
    side?: HeaderSectionSide;
    grow?: boolean;
  };

export const EuiHeaderSection: FunctionComponent<EuiHeaderSectionProps> = ({
  side = 'left',
  children,
  className,
  grow = false,
  ...rest
}) => {
  const classes = classNames(
    'euiHeaderSection',
    {
      'euiHeaderSection--grow': grow,
      'euiHeaderSection--dontGrow': !grow,
    },
    sideToClassNameMap[side],
    className
  );

  return (
    <div className={classes} {...rest}>
      {children}
    </div>
  );
};
