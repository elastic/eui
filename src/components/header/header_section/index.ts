export { EuiHeaderSection } from './header_section';

export {
  EuiHeaderSectionItem,
  EuiHeaderSectionItemProps,
} from './header_section_item';

export { EuiHeaderSectionItemButton } from './header_section_item_button';
