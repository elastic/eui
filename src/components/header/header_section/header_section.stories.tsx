/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import type { Meta, StoryObj } from '@storybook/react';

import { EuiHeader } from '../header';
import { EuiHeaderSection, EuiHeaderSectionProps } from './header_section';

const meta: Meta<EuiHeaderSectionProps> = {
  title: 'EuiHeaderSection',
  component: EuiHeaderSection,
};

export default meta;
type Story = StoryObj<EuiHeaderSectionProps>;

export const Playground: Story = {
  render: ({ ...args }) => (
    <EuiHeader position="fixed">
      <EuiHeaderSection {...args}>Hello world</EuiHeaderSection>
    </EuiHeader>
  ),
};
