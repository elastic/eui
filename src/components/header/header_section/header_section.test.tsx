/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../../test/required_props';

import { EuiHeaderSection } from './header_section';

describe('EuiHeaderSection', () => {
  test('is rendered', () => {
    const component = render(<EuiHeaderSection {...requiredProps} />);

    expect(component).toMatchSnapshot();
  });

  test('renders optional params', () => {
    const component = render(
      <EuiHeaderSection style={{ color: 'blue' }}>
        <span>Some years ago never mind how long precisely...</span>
      </EuiHeaderSection>
    );

    expect(component).toMatchSnapshot();
  });

  describe('grow', () => {
    test('defaults to false', () => {
      const component = render(<EuiHeaderSection />);

      expect(component).toMatchSnapshot();
    });

    test('renders true', () => {
      const component = render(<EuiHeaderSection grow />);

      expect(component).toMatchSnapshot();
    });
  });

  describe('side', () => {
    test('defaults to left', () => {
      const component = render(<EuiHeaderSection />);

      expect(component).toMatchSnapshot();
    });

    test('renders right', () => {
      const component = render(<EuiHeaderSection side="right" />);

      expect(component).toMatchSnapshot();
    });
  });
});
