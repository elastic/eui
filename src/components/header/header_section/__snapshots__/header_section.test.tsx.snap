// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiHeaderSection grow defaults to false 1`] = `
<div
  class="euiHeaderSection euiHeaderSection--dontGrow euiHeaderSection--left"
/>
`;

exports[`EuiHeaderSection grow renders true 1`] = `
<div
  class="euiHeaderSection euiHeaderSection--grow euiHeaderSection--left"
/>
`;

exports[`EuiHeaderSection is rendered 1`] = `
<div
  aria-label="aria-label"
  class="euiHeaderSection euiHeaderSection--dontGrow euiHeaderSection--left testClass1 testClass2"
  data-test-subj="test subject string"
/>
`;

exports[`EuiHeaderSection renders optional params 1`] = `
<div
  class="euiHeaderSection euiHeaderSection--dontGrow euiHeaderSection--left"
  style="color:blue"
>
  <span>
    Some years ago never mind how long precisely...
  </span>
</div>
`;

exports[`EuiHeaderSection side defaults to left 1`] = `
<div
  class="euiHeaderSection euiHeaderSection--dontGrow euiHeaderSection--left"
/>
`;

exports[`EuiHeaderSection side renders right 1`] = `
<div
  class="euiHeaderSection euiHeaderSection--dontGrow euiHeaderSection--right"
/>
`;
